client/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -791,11 +796,25 @@ case class DeltaSharingSource(
       add.id -> add.url
     }.toMap
 
+    // For streaming queries, we return a DataFrame.
+    // The FileIndex here is a one-time use object to list files.
     val params = new RemoteDeltaFileIndexParams(
-      spark, initSnapshot, deltaLog.client.getProfileProvider)
+      spark, initSnapshot, deltaLog.client.getProfileProvider, Some(queryParamsHashId))
     val fileIndex = new RemoteDeltaBatchFileIndex(params, addFilesList)
+
+    val tablePathWithParams =
+      if (ConfUtils.sparkParquetIOCacheEnabled(spark.sessionState.conf)) {
+        // Ensure different query shapes against the same table have distinct entries
+        // in the pre-signed URL cache.
+        QueryUtils.getTablePathWithIdSuffix(
+          params.path.toString, params.queryParamsHashId.get
```

Comment:
nit: you can directly use queryParamsHashId, instead of params.queryParamsHashId.get